### PR TITLE
#171784085 delivers login with social Media

### PR DESCRIPTION
**Pull request by kabundege**
#### What does this PR do?
sets up Social Authentication
 - route
 - configurations
 - middlerware
 - tests
 - controller

 #### Description of Task to be completed
 -  Client (users) relies on social networks’ OAuth 
     for faster authentication when they visit websites

 #### How should this be manually tested?
 > clone the repository, run the server and click the following :
 > - For Google click [Here](http://localhost:5000/api/v1/users/auth/google)
 > - For Facebook click [Here](http://localhost:5000/api/v1/users/auth/facebook)
 #### Any background context you want to provide?
N/A
 #### What are the relevant pivotal tracker stories?
- Find the story [here](https://www.pivotaltracker.com/story/show/171784085)
 #### Screenshots (if appropriate)
> ### Google Strategy
> 
> ### Facebook Strategy
>

### CODE REVIEW

**Review by irfiacre — COMMENTED** [review]
Nice work @kabundege just fix the `tests` and all will be okay.

**Review by fordarnold — COMMENTED** [review]
- Fix the errors in tests.
- Also, when you finish, change the PR label from "WIP" to "Ready for Review".

File: server/controllers/usercontroller.js [hunk 1]
```diff
@@ -61,6 +61,72 @@ class User {
...
+      google = req.user.id;
+    }else{
+      facebook = req.user.id;
+    }
+
+      const loginUser = await userService.findByProp({
+        googleId : `${req.user.id}`,
+        facebookId : `${req.user.id}`,
+      })
+      const exist = await userService.findByProp({
```
Comment (Francois-MUGOROZI) [inline server/controllers/usercontroller.js]: That is good work 
However, here for all the checking for user existence and other validation have to be in middleware. You can contact Fiacre for more about this.

File: server/controllers/usercontroller.js [hunk 2]
```diff
@@ -48,18 +49,94 @@ class User {
...
-      return util.send(res);
+    if (googleSearch[0]||facebookSearch[0]) {
+      if(!googleSearch[0]){
+        facebookSearch[0].dataValues
+      }else{
+        googleSearch[0].dataValues
+    }
+    Action = 'Log In';
+    status = 200;
+      }else if(exist[0]){
```
Comment (irfiacre) [inline server/controllers/usercontroller.js:161]: All checks should be in the middleware

Comment (kabundege) [reply server/controllers/usercontroller.js:161]:
```
That's note a response, it's just setting up 
values that will be sent to the response 
Utility according to the action made.
```

**Review by irfiacre — APPROVED** [review]

**Review by Francois-MUGOROZI — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -21,6 +23,7 @@
     "devtable": " npm run drop && npm run createTables",
     "migration": "babel-node node_modules/.bin/sequelize db:migrate",
     "undomigration": "babel-node node_modules/.bin/sequelize db:migrate:undo:all --to 20200331142748-create-users.js",
+    "test1": "set NODE_ENV=test && npm run drop:test && npm run testTables && nyc --reporter=html --reporter=text mocha --require @babel/register --timeout 5000 ./server/tests/*.test.js --exit",
```
Comment (irfiacre) [inline package.json]: Remove this script

**Review by irfiacre — CHANGES_REQUESTED** [review]

**Review by fordarnold — APPROVED** [review]

**Review by irfiacre — APPROVED** [review]

**Comment by Francois-MUGOROZI** [comment]
Christophe, can you check out those conflicts please?

**Review by bdushimi — APPROVED** [review]

**Pull request merged**